File: expo/android-host/app/src/main/java/com/example/sample/expo/brownfield/host/AppConfig.kt
package com.example.sample.expo.brownfield.host

/** Build-time settings. */
object AppConfig {
  const val DEFAULT_API_BASE_URL = "https://api.github.com"

  /**
   * Where the search screen sends its requests. The benchmark build points
   * this at bench/mock-server through `-PbenchApiBaseUrl`; every other build
   * talks to GitHub.
   */
  val apiBaseUrl: String
    get() = apiBaseUrl(BuildConfig.BENCH_API_BASE_URL)

  fun apiBaseUrl(configured: String): String = configured.ifEmpty { DEFAULT_API_BASE_URL }
}

File: kmp-native-ui/ios-host/HostApp/AppConfig.swift
import Foundation

/// Build-time settings, read from Info.plist.
enum AppConfig {
  static let defaultAPIBaseURL = URL(string: "https://api.github.com")!

  /// Where the search screen sends its requests. The benchmark build points
  /// this at bench/mock-server through the `BENCH_API_BASE_URL` build setting;
  /// every other build talks to GitHub.
  static var apiBaseURL: URL {
    apiBaseURL(from: Bundle.main.object(forInfoDictionaryKey: "BenchAPIBaseURL") as? String)
  }

  static func apiBaseURL(from value: String?) -> URL {
    guard let value, !value.isEmpty, let url = URL(string: value) else {
      return defaultAPIBaseURL
    }
    return url
  }
}
